packages/node-runtime-worker-thread/src/worker-runtime.spec.ts

```diff
@@ -6,13 +6,35 @@ import chai, { expect } from 'chai';
 import chaiAsPromised from 'chai-as-promised';
 import sinonChai from 'sinon-chai';
 import sinon from 'sinon';
-
+import { ObjectId } from 'bson';
+import { inspect } from 'util';
+import { ShellResult } from '@mongosh/shell-evaluator';
 import { startTestServer } from '../../../testing/integration-testing-hooks';
-import { createCaller, exposeAll } from './rpc';
+import { Caller, createCaller, exposeAll } from './rpc';
+import { deserializeEvaluationResult } from './serializer';
+import type { WorkerRuntime } from './worker-runtime';
 
 chai.use(sinonChai);
 chai.use(chaiAsPromised);
 
+const chaiBson: Chai.ChaiPlugin = (chai) => {
+  chai.Assertion.addMethod('bson', function bson(expected) {
+    const obj = this._obj;
+
+    new chai.Assertion(obj).to.be.instanceof(expected.constructor);
+
+    this.assert(
+      inspect(obj) === inspect(expected),
```

Comment:
what is this asserting here?

Reply:
This seems like the easiest way to assert that two bson types are the same:

```js
const { inspect } = require('util');
const { MinKey } = require('bson');

console.log(new MinKey() == new MinKey()); // false
console.log(inspect(new MinKey()) == inspect(new MinKey())); // true
```

Reply:
Rewrote it to use `EJSON.serialize` for comparison, probably slower, but definitely less hacky